src/util/convertHTMLtoJSON.ts

```diff
@@ -0,0 +1,161 @@
+import _ from 'lodash'
+import { Element, HimalayaNode, Text, parse } from 'himalaya'
+import { Block } from '../action-creators/importText'
+
+interface PreBlock extends Block {
+  wasSpan?: boolean,
+  hasSpaces?: boolean,
+}
+
+/** Parses input HTML and saves in JSON array using Himalaya. */
+export const convertHTMLtoJSON = (html: string) => {
+
+  /** Check if string contains only characters from given 'includes' array. */
+  const isStringIncludesOnly = (str: string, includes: string[]) => {
+    const matched =
+      includes
+        .map(char => str.replace(new RegExp(`[^${char}]`, 'g'), '').length)
+        .reduce((acc, cur) => acc + cur, 0)
+    return matched === str.length
+  }
+
+  /** Retrieve attribute from Element node by key. */
+  const getAttribute = (key: string, node: Element) => {
+    const { attributes } = node
+    const attribute = attributes.find(attr => attr.key === key)
+    return attribute ? attribute.value : undefined
+  }
+
+  /** Removes empty nodes and comments from himalaya's JSON output. */
+  const removeEmptyNodesAndComments = (nodes: HimalayaNode[]) => {
+    return nodes.filter(node => {
+      if (node.type === 'element') {
+        if (node.tagName === 'br') return false
+        node.children = removeEmptyNodesAndComments(node.children)
+        return true
+      }
+      if (node.type === 'comment') return false
+      // remove text nodes containing only empty space and new line characters, such nodes could be created by himalaya in case of multi-line html input
+      if (node.type === 'text' && isStringIncludesOnly(node.content, ['\n', ' '])) return false
+      return node.content.length
+    }) as (Element | Text)[]
+  }
+
+  /** Append children to parent as children property if it's necessary. */
```

Comment:
Could you provide more of an explanation of the purpose of `joinChildren`? I don't totally understand.

Reply:
Answered above.